tests/test_matrix_props/test_is_permutation.py

```diff
@@ -0,0 +1,34 @@
+"""Test is_permutation."""
+import numpy as np
+
+from toqito.matrix_props import is_permutation
+
+def test_is_permutation():
+    """Test if matrix is a real permutation matrix."""
+    mat = np.array([[1, 0, 0], [0, 0, 1], [0, 1, 0]])
+    np.testing.assert_equal(is_permutation(mat), True)
+
+def test_is_not_permutation():
+    """Test non-permutation matrix."""
+    mat = np.array([[0, 0, 0, 9], [1, 0, 0, 0], [0, 1, 0, 0], [0,0,1,0]])
+    np.testing.assert_equal(is_permutation(mat), False)
+
+def test_is_not_permutation():

```

Comment:
You can't have two different functions with the same name (`test_is_not_permutation()` exists twice).

Reply:
That was a huge typo on my part. Great catch.  I didn't even notice  when I ran tests that only one was wasn't  executing until you pointed this out.  When I write java, c and other code the compiler catches these errors right away, so I am not used to seeing it stick around in my code this long.  Thanks so much.